src/cryptenroll/cryptenroll.c

```diff
@@ -486,6 +499,28 @@ static int prepare_luks(
                                 strlen(envpw));
                 if (r < 0)
                         return log_error_errno(r, "Password from environment variable $PASSWORD did not work.");
+        } else if(arg_unlock_file) {
+                char *password;
+                size_t password_len;
+
+                r = crypt_keyfile_device_read(
+                                cd,
+                                arg_unlock_file,
+                                &password,
+                                &password_len,
+                                0, 0, 0);
```

Comment:
please use read_full_file() or read_full_file_full() and use `READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE|READ_FULL_FILE_CONNECT_SOCKET`